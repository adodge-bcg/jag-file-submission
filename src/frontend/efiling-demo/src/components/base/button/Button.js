import React from "react";
import PropTypes from "prop-types";
import "./Button.css";

export const Button = ({ onClick, label }) => (
  <div>
    <button className="efile-btn btn" onClick={onClick} type="button">
      {label}
    </button>
  </div>
);

Button.propTypes = {
  onClick: PropTypes.func.isRequired,
  label: PropTypes.string.isRequired
};
