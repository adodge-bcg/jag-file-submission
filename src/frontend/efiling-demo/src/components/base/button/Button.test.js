import React from "react";
import testBasicSnapshot from "../../../TestHelper";

import { Button } from "./Button";

describe("Button Component", () => {
  const onClick = jest.fn();
  const label = "button label";

  test("Button matches the snapshot", () => {
    const buttonComponent = <Button onClick={onClick} label={label} />;

    testBasicSnapshot(buttonComponent);
  });
});
