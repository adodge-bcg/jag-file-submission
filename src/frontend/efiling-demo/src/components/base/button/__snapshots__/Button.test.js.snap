// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component Button matches the snapshot 1`] = `
<div>
  <button
    className="efile-btn btn"
    onClick={[MockFunction]}
    type="button"
  >
    button label
  </button>
</div>
`;
